src/buildstream_plugins/sources/cargo.py

```diff
@@ -234,9 +233,8 @@ def _download(self, url):
     #    (str): The URL for this crate
     #
     def _get_url(self, alias=None):
-        url = self.cargo.translate_url(self.cargo.url, alias_override=alias)
         path = "{name}/{name}-{version}.crate".format(name=self.name, version=self.version)
-        return urljoin(f"{url}/", path)
+        return self.cargo.translate_url(self.cargo.url, suffix=path, alias_override=alias)
```

Comment:
Is this missing a `/` separator?

Reply:
Yes and no. The whole `urljoin` dance was supposed to allow specifying the url with and without the trailing slash.

However if we're working with an "untranslated" url (i.e. with an alias), then 1. `urljoin` doesn't work and 2. we might be unable to know whether the url provided has a trailing slash or not. I'm inclined to assume the trailing slash and not include it.

#52 is where this was changed, and earlier we were assuming the url didn't include a trailing slash.

So it comes down to determining what we think would be most correct and go with it.

Reply:
Requiring aliases to end with a trailing slash does sound sensible, although I'm not sure whether this is a concern with regards to backward compatibility.

We also need to update the default URL in `configure()` and the doc comment.

And update `BST_MIN_VERSION` to `2.2`.

Reply:
Updated, thanks.

Regarding `BST_MIN_VERSION`, I didn't update it because until 2.2 is released it wouldn't work on buildstream. I'm not sure what's the best way to do it, but we might want to delay merging this until 2.2 is ready.

Reply:
May it be reasonable to tag the buildstream repository as 2.2.0.dev0 as soon as https://github.com/apache/buildstream/pull/1899 is merged, and then we merge this (and the bst-plugins-experimental MR) with `BST_MIN_VERSION` set to `2.2`?